src/controllers/activegate/reconciler/automaticapimonitoring/reconciler.go

```diff
@@ -0,0 +1,89 @@
+package automaticapimonitoring
+
+import (
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/pkg/errors"
+)
+
+type AutomaticApiMonitoringReconciler struct {
+	dtc            dtclient.Client
+	name           string
+	kubeSystemUUID string
+}
+
+func NewReconciler(dtc dtclient.Client, name, kubeSystemUUID string) *AutomaticApiMonitoringReconciler {
+	return &AutomaticApiMonitoringReconciler{
+		dtc,
+		name,
+		kubeSystemUUID,
+	}
+}
+
+func (r *AutomaticApiMonitoringReconciler) Reconcile() error {
+	objectID, err := r.ensureSettingExists()
+
+	if err != nil {
+		return err
+	}
+
+	if objectID != "" {
+		log.Info(fmt.Sprintf("created setting '%s' for Cluster '%s'. Settings object ID: %s", r.name, r.kubeSystemUUID, objectID))
+	} else {
+		log.Info(fmt.Sprintf("setting '%s' for Cluster '%s' already exists.", r.name, r.kubeSystemUUID))
+	}
+
+	return nil
+}
+
+func (r *AutomaticApiMonitoringReconciler) ensureSettingExists() (string, error) {
+	if r.name == "" {
```

Comment:
checking the `r.name != ""` is not necessary, because its the name of the `DynaKube`, and it must have a name to exist in Kubernetes